File: react_native_app/src/features/productos/domain/usecases/updateProducto.ts
import { UseCase } from '../../../../core/usecase/usecase';
import { Producto } from '../entities/Producto';
import { ProductoRepository } from '../repositories/ProductoRepository';

/** Actualiza un producto (PUT a la API) y refleja el cambio en memoria. */
export class UpdateProducto implements UseCase<Producto, Producto> {
  constructor(private readonly repository: ProductoRepository) {}

  call(producto: Producto): Promise<Producto> {
    return this.repository.updateProducto(producto);
  }
}

File: flutter_app/lib/features/productos/domain/usecases/update_producto.dart
import '../../../../core/usecase/usecase.dart';
import '../entities/producto.dart';
import '../repositories/producto_repository.dart';

/// Actualiza un producto (PUT a la API) y refleja el cambio en memoria.
class UpdateProducto implements UseCase<Producto, Producto> {
  final ProductoRepository repository;
  UpdateProducto(this.repository);

  @override
  Future<Producto> call(Producto producto) => repository.updateProducto(producto);
}
